vault/data_source_kubernetes_auth_backend_role.go

```diff
@@ -84,6 +84,12 @@ func kubernetesAuthBackendRoleDataSource() *schema.Resource {
 			},
 			Deprecated: "use `token_bound_cidrs` instead if you are running Vault >= 1.2",
 		},
+		"audience": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Default:     "",
+			Description: "Optional Audience claim to verify in the JWT.",
```

Comment:
[Tokens can have several audiences.](https://sourcegraph.com/github.com/kubernetes/api@master/-/blob/authentication/v1/types.go#L143-L149) Did you consider allowing a set of audiences here, mandating at least one intersection with the audience set in the token?

That is, you could say that you'll admit audiences "a1" and "a2," and receive a token with its audiences bound to "a2" and "a3." That would be a match, since the set intersection is nonempty.

Reply:
Hi @seh ! Thanks for pointing that out!

I'm noticing your link leads to the Kubernetes API. This code points at the _Vault Kubernetes auth plugin's_ API. The doc regarding this individual field is located [here](https://www.vaultproject.io/api/auth/kubernetes/index.html#audience) with its particular schema being driven off of the plugin [here](https://github.com/hashicorp/vault-plugin-auth-kubernetes/blob/master/path_role.go#L48-L51). So I'm thinking the code should be good to go as-is.